test/external_unfocused-shared-param_not-inherited-by-focus.xspec

```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<x:description run-as="external" stylesheet="external_mirror.xsl" xmlns:mirror="x-urn:test:mirror"
+	xmlns:x="http://www.jenitennison.com/xslt/xspec">
+
+	<!--
+		This test is based on external_unfocused-param_not-inherited-by-focus.xspec, except that
+		the parameter is referenced via x:like. The structure mentioned next is the structure
+		after resolving x:like.
+
+		Test //x:scenario/x:param when they are unfocused by @focus on a non-inner scenario.
+	-->
+
+	<x:scenario focus="focus" label="Focused scenario" />
+
+	<x:scenario label="Scenario parameter" shared="yes">
+		<x:param name="unfocused-scenario-param" select="error() div 0 + $bogus:nonexistent" />
+	</x:scenario>
+
+	<x:scenario label="Unfocused scenario">
+		<x:call function="Q{bogus}nonexistent" />
+		<x:like label="Scenario parameter" />
```

Comment:
After resolving `x:like`, this scenario looks like this:
```xspec
<x:scenario>
  <x:call />
  <x:param />
  <x:expect />
</x:scenario>
```
which violates the schema.

Unfortunately we can't just swap `x:call` and `x:like`, because the current schema does not allow `x:like` before `x:call`.

So what we can do at this time would be
```xspec
<x:scenario>
  <x:like />
  <x:scenario>
    <x:call />
    <x:expect />
  </x:scenario>
</x:scenario>
```

Reply:
Good catch about the structure not being valid after `x:like` is resolved. I fixed that in 6358116de24a7b781cb560ee02e0d4dc55cf0fed.

> the current schema does not allow `x:like` before `x:call`.

I had noticed that, too, and I wondered if the schema should allow it. The only disadvantage I can think of is that it would be even easier to create instances where a test isn't valid after `x:like` is resolved, as in my earlier mistake. We don't have a good way to alert users to such mistakes, do we? Do you think I should log an issue about relaxing this schema restriction and/or creating a way to help users detect when resolving `x:like` makes the test invalid?

Reply:
> Good catch about the structure not being valid after `x:like` is resolved. I fixed that in [6358116](https://github.com/xspec/xspec/commit/6358116de24a7b781cb560ee02e0d4dc55cf0fed).

LGTM.

> I had noticed that, too, and I wondered if the schema should allow it. The only disadvantage I can think of is that it would be even easier to create instances where a test isn't valid after `x:like` is resolved, as in my earlier mistake. We don't have a good way to alert users to such mistakes, do we? Do you think I should log an issue about relaxing this schema restriction and/or creating a way to help users detect when resolving `x:like` makes the test invalid?

Added a comment to a related issue: https://github.com/xspec/xspec/issues/477#issuecomment-852685754
Detecting invalid documents after resolving `x:like` (and `x:import`) would be very useful. Unfortunately, it will not be a simple task.

Reply:
Thanks for adding the comment there so it can be considered at some point.